### PR TITLE
Add/disable control on driver input

### PR DESCRIPTION
**Pull request by petebrink**

### CODE REVIEW

File: firmware/steering/kia_soul_ps/current_control_state.h [hunk 1]
```diff
@@ -73,5 +73,5 @@ typedef struct
     double steering_angle_last; /* Last steering angle recorded */
     //
     //
-    uint32_t timestamp_us; /* Keeps track of last control loop time in us */
+    uint32_t timestamp_ms; /* Keeps track of last control loop time in us */
```
Comment (ltbj) [inline firmware/steering/kia_soul_ps/current_control_state.h]: Comment needs to be updated to match the new variable name and unit.

File: firmware/steering/kia_soul_ps/steering_control_module.ino [hunk 2]
```diff
@@ -121,69 +120,71 @@ static PID pid_params;
...
+// Parameters:  [in/out] timer - the timer to update
+//              [in] period - period of the timer
 // 
 // *****************************************************
-static uint32_t timer_delta_us( uint32_t last_time, uint32_t* current_time )
+static void update_periodic_timer( uint32_t* const timer, uint32_t period )
 {
-    uint32_t delta = 0;
-    uint32_t local_time = micros( );
+    *timer += period;
```
Comment (ltbj) [inline firmware/steering/kia_soul_ps/steering_control_module.ino]: Does this pointer need to be validated before being modified?

File: firmware/steering/kia_soul_ps/steering_control_module.ino [hunk 3]
```diff
@@ -428,7 +435,7 @@ bool check_driver_steering_override( )
 //              [out] torque_spoof - structure containing the integer torque values
 // 
 // *****************************************************
-void calculate_torque_spoof( float torque, struct torque_spoof_t* spoof ) 
+void calculate_torque_spoof( float torque, struct torque_spoof_s* spoof )
```
Comment (ltbj) [inline firmware/steering/kia_soul_ps/steering_control_module.ino]: The `spoof` pointer needs to be checked before it's referenced.

**Comment by LucasBuckland** [comment]
Has this been tested on the car? If so then I'm good to approve.

**Pull request closed without merging**